src/scripts/main.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
+const form = document.querySelector('form');
+
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (element) {
+      element.addEventListener(eventName, () => {
+        const string = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+        resolve(string);
+      });
+    } else{
+      reject(new Error('Element doesn\'t exist'));
+    }
+  });
 }
 
+const messageDiv = document.createElement('div');
```

Comment:
The creation of elements and adding them to the DOM is usually done in the HTML file or in a separate function. Consider moving this to a separate function for better code organization.